packages/browser/src/Session.ts

```diff
@@ -227,6 +227,27 @@ export class Session extends EventEmitter {
         this
       )
     );
+
+    // When a session is logged in, we want to track its ID in local storage to
+    // enable silent refresh.
+    this.on("login", () => {
+      // Store the current session ID specifically in 'localStorage' (i.e., not using
+      // any other storage mechanism), as we don't deem this information to be
+      // sensitive, and we want to ensure it survives a browser tab refresh.
+      window.localStorage.setItem(KEY_CURRENT_SESSION, this.info.sessionId);
+    });
+
+    // On logout, silent refresh should no longer be considered a viable option.
+    this.on("logout", () => {
```

Comment:
Question: do we reliably have a "session expired" event? If so, we should also remove there which might assist in preventing the `client_invalid` errors

Reply:
That's a good point, and we do. It may save a round trip in some cases, even though the event being thrown is quite unlikely: either the browser tab where the refresh happens is in focus, and refresh happens automatically in the background, or it is out of focus, the refresh doesn't happen in the background, but sending the session expired signal doesn't happen either (or at least this is my current understanding). 

Regardless, session expiration is a possible event, and it should be handled here.

Reply:
hmm, is there maybe a way of tracking when a refresh should've happened, and if it didn't on refocus of the tab (https://developer.mozilla.org/en-US/docs/Web/API/Page_Visibility_API) we can detect that the session has potentially expired and we trigger a "refresh the session now", and then clear if it's returning `interaction_required`?

Reply:
Fixed in cadb21e7095c3d020f29d6146c7cd7e5ae1561fa and 124c004311e95d3de470bc408732cfc469cb1d80